domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClientStreamResult.java

```diff
@@ -0,0 +1,120 @@
+package org.folio.rest.persist;
+
+import io.vertx.core.Handler;
+import io.vertx.core.streams.ReadStream;
+import org.folio.rest.jaxrs.model.ResultInfo;
+
+/**
+ * The result of successful completion of PostgresCLient.streamGet
+ *
+ * @param <T> each item returned in stream is of this type
+ */
+class PostgresClientStreamResult<T> implements ReadStream<T> {
+
+  private final ResultInfo resultInfo;
+
+  private Handler<T> streamHandler;
+  private Handler<Void> endHandler;
+  private Handler<Throwable> exceptionHandler;
+
+  /**
+   * Only to be constructed from PostgresClient itself
+   *
+   * @param resultInfo
+   */
+  PostgresClientStreamResult(ResultInfo resultInfo) {
+    this.resultInfo = resultInfo;
+  }
+
+  /**
+   * The ResultInfo for the streamGet.
+   *
+   * @return Result information.. including totalRecords and facets
+   */
+  public ResultInfo resultInto() {
+    return this.resultInfo;
+  }
+
+  /**
+   * Sets stream handler for each returned item of type T
+   *
+   * @param streamHandler
+   * @return this instance (fluent)
+   */
+  @Override
+  public PostgresClientStreamResult<T> handler(Handler<T> streamHandler) {
+    this.streamHandler = streamHandler;
+    return this;
+  }
+
+  /**
+   * Sets handler to be called when stream is complete
+   *
+   * @param endHandler
+   * @return this instance (fluent)
+   */
+  @Override
+  public PostgresClientStreamResult<T> endHandler(Handler<Void> endHandler) {
+    this.endHandler = endHandler;
+    return this;
+  }
+
+  /**
+   * Sets handler to be called if an exception occurs Both exceptions in user
+   * handlers and PostgresClient are caught.
```

Comment:
Can you add that after the exception has been called neither the item handler nor the end handler will be called?